### ISSUE TITLE
Exception handler for annotated services

### ISSUE THREAD

**Issue by hyangtack** [line/armeria#699]
Currently there is no way to handle an exception raised from an annotated service method. It would be useful if we introduce a simple exception handling mechanism like spring `@ExceptionHandler` annotation.

**Comment by trustin** [comment]
@yuinacor You had some opinion on this before. Any suggestions?

**Comment by hyangtack** [comment]
Also, no way to handle `null` which is returned from the service method, because the converters are based on the type of class defined by a user. Need to figure out a better way to handle responses.

**Comment by kojilin** [comment]
```
public class Service {

  @Converter(JsonConverter.class)
  @ExceptionHandler({IOException.class})
  public CompletableFuture<Model> handle(..){
     ....
  }
}
```

so we need ExceptionHandler
handle method can return `Completable<T>` or `T`. `T` can be type that converter can handle HttpResponse/AggregatedHttpMessage/converter target.

parameter can handle ServiceRequestContext, httpRequest, AggregatedHttpMessage, HttpParameters. No `@Param` or  `@Path`.

How do you think?

I also want to add annotation like `@ResponseStatus(404, json)`. But looks like it doesn't works well with converter. So give up this idea.

**Comment by hyangtack** [comment]
@kojilin Basically I agree with you. 👍 I think we need to do dog-fooding your implementation to figure out more convenient way than Spring's exception handling mechanism. We may have better idea after that.

**Comment by trustin** [comment]
@yuinacor Any comments?

**Comment by hyangtack** [comment]
@kojilin Hello! Do you have any progress on this issue? 😃 If you didn't start writing code yet, could I take this issue? I think I can implement a simple exception handler with a similar idea to `@Decorate` annotation implementation.

**Comment by kojilin** [comment]
Ah, thanks, please

**Comment by hyangtack** [comment]
Merged #847 . Close this issue.
